### ISSUE TITLE
[Phase 1][colocation] Set colocation_id consistently in schemas

### ISSUE THREAD

**Issue by jaki** [yugabyte/yugabyte-db#5017]
Jira Link: [[DB-376]](https://yugabyte.atlassian.net/browse/DB-376)
`pgtable_id` (now `colocation_id`) for schemas are haphazardly set.

- Create table request from postgres does not set it
- `CatalogManager::CreateTable` does not set it
- `RaftGroupMetadata::AddTable` does set it
- `CatalogManager::AlterTable` does set it
- `CatalogManager::AddIndexInfoToTable` does not set it
- `CatalogManager::MarkIndexInfoFromTableForDeletion` does not set it
- `RaftGroupMetadata::SetSchema` does set it (since commit d92b23461a81dee15446b356acdc0ae2440f4a5f)

Ideally, `pgtable_id` would be set from wherever it originates from: postgres create table and alter table requests to master.  That way, we don't have to set it ad hoc on master and tserver.  However, this may cause backwards compatibility issues.

Another option is to set it from master: `CatalogManager::CreateTable` and ((`CatalogManager::AddIndexInfoToTable` and `CatalogManager::MarkIndexInfoFromTableForDeletion`) or `AsyncAlterTable::SendRequest`).  This enables us to get rid of the ad hoc setting at tserver.

[DB-376]: https://yugabyte.atlassian.net/browse/DB-376?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by siddharth2411** [comment]
As part of this issue, we essentially want to make sure that colocation_id do not get modified. To do this, we want to put checks on colocation_id wherever it can possibly get changed. There might be some other methods than the ones mentioned above which may change it.  

- Verify all the PB in `CatalogManager` that store colocation_id and ensure none of the function that use these PBs are modifying the colocation_id.
- Verify that colocation_id is being set (correctly) only once in `RaftGroupMetadata`.

**Comment by frozenspider** [comment]
Current state of things:
* `CatalogManager::CreateTable` does set `colocation_id` - which makes sense since it's an entry point for all table creation.
* `RaftGroupMetadata::AddTable` does not set it, but it does verify `colocation_id` remains unchanged
* `CatalogManager::AlterTable` does not set it - `AlterTableRequestPB` has no field for that
* `CatalogManager::AddIndexInfoToTable` does not set it, for the same reasons. Moreover, it's called from `CatalogManager::CreateTable`
* `CatalogManager::MarkIndexInfoFromTableForDeletion` does not set it
* `RaftGroupMetadata::SetSchema` does not set it, but it does verify `colocation_id` remains unchanged

To summarize, `colocation_id` is only set in `CatalogManager::CreateTable` which is consistent. Closing this issue.
